test/fail_compilation/fail2789.d

```diff
@@ -53,7 +47,7 @@ void f4() {}
 void f4() {}    // conflict
 
 void f5() @safe {}
-void f5() @system {}    // conflict
+void f5() @system {}    // no conflict because of attribute based overloading in in extern(D)
```

Comment:
Yes, `extern(D)` preserves the mangling, but does overload resolution work well, or the user has to resort to hackery (e.g. [externDFunc](https://github.com/dlang/druntime/blob/v2.094.2/src/core/internal/traits.d#L122) to traits)? If the answer is no, I think it's not a bad idea to deprecate such form of likely accidental overloading.

Reply:
Using `__traits(getOverloads, ...)` should also work IIRC.
But thats out of scope for this PR.

Reply:
I think that is kind of a controversial change. I don't really know if D is able to do overload resolution based on attributes. This fails to compile on run.dlang.io:

```d
void func() pure {}
void func() @nogc {}

void main() @nogc
{
    func();
}
```